watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination_withRO.py

```diff
@@ -0,0 +1,562 @@
+        default={
+            "flowsheet_costing_block": m.fs.ro_costing,
+            "costing_method_arguments": {"cost_electricity_flow": False},
+        }
+    )
+
+    # Aggregate unit level costs and calculate overall process costs
+    m.fs.zo_costing.cost_process()
+    m.fs.ro_costing.cost_process()
+
+    # Annual disposal of brine
+    m.fs.brine_disposal_cost = Expression(
+        expr=(
+            m.fs.zo_costing.utilization_factor
+            * (
+                m.fs.zo_costing.waste_disposal_cost
+                * pyunits.convert(
+                    m.fs.brine.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / m.fs.zo_costing.base_period,
+                )
+            )
+        ),
+        doc="Cost of disposing of brine waste",
+    )
+
+    m.fs.dye_recovery_revenue = Expression(
+        expr=(
+            m.fs.zo_costing.utilization_factor
+            * m.fs.zo_costing.dye_mass_cost
+            * pyunits.convert(
+                m.fs.dye_retentate.flow_mass_comp[0, "dye"],
+                to_units=pyunits.kg / m.fs.zo_costing.base_period,
+            )
+        ),
+        doc="Savings from dye recovered back to the plant",
+    )
+
+    m.fs.water_recovery_revenue = Expression(
+        expr=(
+            m.fs.zo_costing.utilization_factor
+            * m.fs.zo_costing.recovered_water_cost
+            * pyunits.convert(
+                m.fs.permeate.properties[0].flow_vol,
+                to_units=pyunits.m**3 / m.fs.zo_costing.base_period,
+            )
+        ),
+        doc="Savings from water recovered back to the plant",
+    )
+
+    # Combine results from costing packages and calculate overall metrics
+    @m.Expression()
+    def total_capital_cost(b):
+        return (
+            pyunits.convert(
+                m.fs.zo_costing.total_capital_cost, to_units=pyunits.USD_2018
+            )
+            + m.fs.ro_costing.total_investment_cost
+        )
+
+    @m.Expression()
+    def total_operating_cost(b):
+        return (
+            pyunits.convert(
+                m.fs.zo_costing.total_fixed_operating_cost,
+                to_units=pyunits.USD_2018 / pyunits.year,
+            )
+            + pyunits.convert(
+                m.fs.zo_costing.total_variable_operating_cost,
+                to_units=pyunits.USD_2018 / pyunits.year,
+            )
+            + m.fs.ro_costing.total_operating_cost
+        )
+
+    @m.Expression()
+    def total_externalities(b):
+        return pyunits.convert(
+            m.fs.water_recovery_revenue
+            + m.fs.dye_recovery_revenue
+            - m.fs.brine_disposal_cost,
+            to_units=pyunits.USD_2018 / pyunits.year,
+        )
+
+    @m.Expression()
+    def LCOT(b, doc="Levelized cost of treatment with respect to volumetric feed flow"):
+        return (
+            b.total_capital_cost * b.fs.zo_costing.capital_recovery_factor
+            + b.total_operating_cost
+            - b.total_externalities
```

Comment:
I think this makes sense now (regarding the value of recovered water), if we are considering the LCOT w/respect to the manufacturing plant.